src/components/Cart/BillingForm.tsx

```diff
@@ -0,0 +1,168 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Controller, useForm } from "react-hook-form";
+import { billingValidationShema } from "./validation";
+import FormField from "@components/FormComponents/FormField";
+import { Button } from "@components/Button/Button";
+import { ButtonVariant } from "../../main/types/enums";
+import { Autocomplete } from "@components/FormComponents/Autocomplete";
+import { FormControl } from "@mui/material";
+import { twMerge } from "tailwind-merge";
+import Label from "@components/FormComponents/Label";
+import { useEffect, useMemo } from "react";
+import { AppDispatch } from "../../redux/app/store";
+import { useDispatch } from "react-redux";
+import {
+  getCities,
+  getCountries,
+} from "../../redux/features/location/locationSlice";
+import { useAppSelector } from "../../redux/app/hooks";
+import {
+  transformCitiesToOptions,
+  transformCountriesToOptions,
+} from "../../main/helpers";
+import {
+  selectCities,
+  selectCountries,
+} from "../../redux/features/location/selectors";
+
+export const BillingForm = () => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    control,
+    watch,
+    setValue,
+  } = useForm({
+    resolver: yupResolver(billingValidationShema),
+  });
+
+  const country = watch("country");
+  const dispatch: AppDispatch = useDispatch();
+  const countries = useAppSelector(selectCountries);
+  const cities = useAppSelector(selectCities);
+
+  useEffect(() => {
+    dispatch(getCountries());
+  }, []);
+
+  useEffect(() => {
+    if (country) {
+      dispatch(getCities({ country: country }));
+    } else {
+      setValue("city", "");
+    }
+  }, [country]);
+
+  const onSubmit = async () => {};
+
+  const countriesOptions = useMemo(
+    () => transformCountriesToOptions(countries),
+    [countries],
+  );
+
+  const citiesOptions = useMemo(
+    () => transformCitiesToOptions(cities),
+    [cities],
+  );
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="flex flex-col gap-7 max-w-[500px]"
+    >
+      <div className="flex flex-col gap-4">
+        <FormField
+          label="First name"
+          placeholder="firstName"
```

Comment:
Do you really need placeholders in camel case? It is better to have like a label

Reply:
done